src/app/profile/payment-methods/payment-methods.component.js

```diff
@@ -0,0 +1,140 @@
+import angular from 'angular';
+import template from './payment-methods.tpl';
+import recurringGiftsComponent from './recurring-gifts/recurring-gifts.component';
+import profileService from 'common/services/api/profile.service.js';
+import paymentMethod from './payment-method/payment-method.component';
+import paymentMethodFormModal from 'common/components/paymentMethods/paymentMethodForm/paymentMethodForm.modal.component';
+import loadingOverlay from 'common/components/loadingOverlay/loadingOverlay.component';
+import giveModalWindowTemplate from 'common/templates/giveModalWindow.tpl';
+import paymentMethodDisplay from 'common/components/paymentMethods/paymentMethodDisplay.component';
+import sessionEnforcerService from 'common/services/session/sessionEnforcer.service';
+import {Roles} from 'common/services/session/session.service';
+
+class PaymentMethodsController {
+
+  /* @ngInject */
+  constructor($uibModal, profileService, sessionEnforcerService, $log, $timeout, $window, $location) {
+    this.log = $log;
+    this.$uibModal = $uibModal;
+    this.paymentMethod = 'bankAccount';
+    this.profileService = profileService;
+    this.loading = false;
+    this.submissionError = {error: ''};
+    this.successMessage = { show: false };
+    this.$timeout = $timeout;
+    this.$window = $window;
+    this.$location = $location;
+    this.sessionEnforcerService = sessionEnforcerService;
+  }
+
+  $onDestroy(){
+    if(this.paymentMethodFormModal) {
+      this.paymentMethodFormModal.close();
+    }
+  }
+
+  $onInit(){
+    this.enforcerId = this.sessionEnforcerService([Roles.registered], {
+      'sign-in': () => {
+        this.loading = true;
+        this.loadPaymentMethods();
+        this.loadDonorDetails();
+      },
+      cancel: () => {
+        this.$window.location = '/';
+      }
+    });
+    this.loading = true;
+    this.loadPaymentMethods();
+    this.loadDonorDetails();
+  }
+
+  loadDonorDetails() {
+    this.profileService.getDonorDetails()
+      .subscribe((data) => {
+        this.mailingAddress = data.mailingAddress;
+      });
+  }
+
+  loadPaymentMethods() {
+    this.profileService.getPaymentMethodsWithDonations()
+      .subscribe(
+        data => {
+          this.loading = false;
+          this.paymentMethods = data;
+        },
+        error => {
+          this.loading = false;
+          this.error = 'Failed retrieving payment methods.';
```

Comment:
This doesn't go away if the session enforcer loads payment methods again. You need to clear it out on success.

Also I think it should go in a bootstrap alert.